File: server/src/models/Analytic.ts
import {
  Column,
  DataType,
  Default,
  IsUUID,
  Model,
  PrimaryKey,
  Table,
} from "sequelize-typescript";

@Table({
  timestamps: true,
  underscored: true,
})
export class Analytic extends Model {
  @IsUUID(4)
  @PrimaryKey
  @Default(DataType.UUIDV4)
  @Column
  id: string;

  @Column
  ip: string;

  @Column
  useragent: string;

  @Column
  page: string;

  @Column
  modal: string;

  @IsUUID(4)
  @Column
  modalId: string;

  @Column
  token: string;

  @IsUUID(4)
  @Column
  userId: string;

  @Column(DataType.JSONB)
  query: Record<string, string | string[]>;
}

File: go-server/internal/models/analytic.go
package models

import "gorm.io/datatypes"

func init() {
	registerModel(&Analytic{})
}

// Analytic isn't a GraphQL type at all on the Node side — no @ObjectType(),
// no @Field() anywhere on the model. It's purely an internal event log
// written by GeneralResolver.ping (graph/general.resolvers.go).
type Analytic struct {
	Base

	IP        string
	UserAgent string `gorm:"type:varchar(510)"`
	Page      string
	Modal     string
	ModalID   string
	Token     string
	UserID    string
	Query     datatypes.JSON `gorm:"type:jsonb"`
}
